src/components/dashboard/Reason.js

```diff
@@ -249,7 +238,6 @@ const SendReason = (props: AmountProps) => {
             placeholder="Add a message"
             placeholderTextColor={theme.colors.darkGray}
             enablesReturnKeyAutomatically
-            onSubmitEditing={next}
```

Comment:
Actually it's a good option to go next on the "Done" press.
Please revert changes and add `isDisabledNextButton` flag check to the `next()` function instead:

```
  if (isDisabledNextButton !== false) {
    return
  }

  const [nextRoute, ...nextRoutes] = screenState.nextRoutes || []

  props.screenProps.push(nextRoute, {
// ...
```

Also, i see the route & params you prepare in the `next()` function are the (almost?) same that are passing to the `<NextButton>`. So it would be good to reuse those values (e.g. calculate them via `useMemo`)